subsys/logging/log_backend_rtt.c

```diff
@@ -159,8 +163,12 @@ static int log_backend_rtt_write(void)
 {
 	if (IS_ENABLED(CONFIG_LOG_BACKEND_RTT_MODE_BLOCK)) {
 		return log_backend_rtt_write_block();
-	} else if (IS_ENABLED(CONFIG_LOG_BACKEND_RTT_MODE_DROP)) {
+	} else {
```

Comment:
can you use IS_ENABLED here as well?

Reply:
Do you want to see this code:

```
} else if (IS_ENABLED(CONFIG_LOG_BACKEND_RTT_MODE_DROP)) {
#if CONFIG_LOG_BACKEND_RTT_MODE_DROP
	return log_backend_rtt_write_drop();
#else
	return 0;
#endif
} else {
	return 0;
}
```

I think GCC does not optimize/remove the call to `log_backend_rtt_write_drop()` if `IS_ENABLED(CONFIG_LOG_BACKEND_RTT_MODE_DROP)` is false. So we still need to define the function `log_backend_rtt_write_drop()` even if we do not use it.

Reply:
GCC will remove without a warning code if it can be resolved during compilation.

Reply:
My build issue started with `DROP_MAX` not being defined because its definition is protected by `#if CONFIG_LOG_BACKEND_RTT_MODE_DROP`. So if you really want to keep `IS_ENABLED(CONFIG_LOG_BACKEND_RTT_MODE_DROP)` then we need to remove the first `#if CONFIG_LOG_BACKEND_RTT_MODE_DROP`

Reply:
ah, in that case i usually do on top of the file
```
ifndef CONFIG_CUT_BY_DISABLED_FEATURE
#define CONFIG_CUT_BY_DISABLED_FEATURE 0
#endif 
```
I think this approach makes it clearer keeping source code without #ifdefs